Sunrise/ViewModels/StoreSelectionViewModel.swift

```diff
@@ -0,0 +1,263 @@
+        return channels.count + (expandedChannelIndexPath.value != nil ? 1 : 0)
+    }
+
+    func storeNameAtIndexPath(indexPath: NSIndexPath) -> String {
+        return channels[rowForChannelAtIndexPath(indexPath)].name?.localizedString ?? ""
+    }
+
+    func storeImageUrlAtIndexPath(indexPath: NSIndexPath) -> String {
+        return channels[rowForChannelAtIndexPath(indexPath)].details?.imageUrl ?? ""
+    }
+
+    func expansionTextAtIndexPath(indexPath: NSIndexPath) -> String {
+        if indexPath == expandedChannelIndexPath.value {
+            return NSLocalizedString("Less info", comment: "Less info")
+        } else {
+            return NSLocalizedString("More info", comment: "More info")
+        }
+    }
+
+    func storeDistanceAtIndexPath(indexPath: NSIndexPath) -> String {
+        let channel = channels[rowForChannelAtIndexPath(indexPath)]
+
+        if let userLocation = userLocation.value, lat = channel.details?.lat, lon = channel.details?.lon {
+            let channelLocation = CLLocation(latitude: lat, longitude: lon)
+            let distance = userLocation.distanceFromLocation(channelLocation)
+            return String(format: "%.1f", arguments: [distance / 1000]) + " km"
+        }
+        return "-"
+    }
+
+    func availabilityAtIndexPath(indexPath: NSIndexPath) -> String {
+        let quantity = 3
+
+        switch quantity {
+        case 0:
+            return NSLocalizedString("This item is currently not available", comment: "Item not available")
+        case 1..<3:
+            return NSLocalizedString("Hurry up! Only few items left", comment: "Few items left")
+        default:
+            return NSLocalizedString("This item is available", comment: "Item available")
+        }
+    }
+
+    func availabilityColorAtIndexPath(indexPath: NSIndexPath) -> UIColor {
+        let quantity = 3
+
+        switch quantity {
+        case 0:
+            return UIColor(red:1.00, green:0.00, blue:0.00, alpha:1.0)
+        case 1..<3:
+            return UIColor(red:1.00, green:0.46, blue:0.10, alpha:1.0)
+        default:
+            return UIColor(red:0.55, green:0.78, blue:0.25, alpha:1.0)
+        }
+    }
+
+    private func rowForChannelAtIndexPath(indexPath: NSIndexPath) -> Int {
+        var channelRow = indexPath.row
+        if let expandedRow = expandedChannelIndexPath.value?.row where channelRow > expandedRow {
+            channelRow -= 1
+        }
+        return channelRow
+    }
+
+    // MARK: - Creating a reservation
+
+    private func reserveProductVariant() {
+
+    }
+
+    private func reserveProductVariant(channel: Channel) -> SignalProducer<Void, NSError> {
+        return SignalProducer { observer, disposable in
+            let selectedChannel = ["typeId": "channel", "id": channel.id ?? ""]
+            let lineItemDraft: [String: AnyObject] = ["productId": self.product.id ?? "",
+                                                      "variantId": self.currentVariantId() ?? 1,
```

Comment:
The assumption probably is that the master variant always has the id `1`, but this may not be the case (e.g. due to http://dev.commercetools.com/http-api-projects-products.html#change-master-variant).

Generally, I would prefer it if the app does not knowingly send an invalid request (e.g. Channel or Product ID is missing) but rather branches into an else block with a more appropriate error message, should any of those values be `null`.